packages/oscal-react-library/src/components/oscal-utils/OSCALMediaTypeUtils.ts

```diff
@@ -0,0 +1,334 @@
+          return "Microsoft Word (.docx)";
+        case "vnd.ms-excel":
+          return "Microsoft Excel (.xls)";
+        case "vnd.openxmlformats-officedocument.spreadsheetml.sheet":
+          return "Microsoft Excel (.xlsx)";
+        case "vnd.ms-powerpoint":
+          return "Microsoft PowerPoint (.ppt)";
+        case "vnd.openxmlformats-officedocument.presentationml.presentation":
+          return "Microsoft PowerPoint (.pptx)";
+      }
+      return undefined;
+    },
+
+    (type) => {
+      // Note: Support for the YAML media type and suffix is currently a draft RFC. We support
+      // it because it is already in semi-common use and a specific document type for OSCAL.
+      // https://www.ietf.org/id/draft-ietf-httpapi-yaml-mediatypes-06.html
+      switch (type.subtype) {
+        case "yaml":
+        case "x-yaml":
+          return "YAML";
+        case "json":
+          return "JSON";
+        case "xml":
+          return "XML";
+      }
+      // Fallback to the base document type for some specific suffixes.
+      switch (type.suffix) {
+        case Suffix.JSON:
+          return "JSON";
+        case Suffix.XML:
+          return "XML";
+        case "yaml":
+          return "YAML";
+      }
+    },
+    (type) => (type.subtype === "pdf" ? "PDF" : undefined),
+    (type) => (type.subtype === "zip" ? "ZIP" : undefined),
+    (type) => (type.subtype === "pem-certificate-chain" ? "PEM Chain" : undefined),
+    (type) => (type.subtype === "pkcs8" ? "PKCS8" : undefined),
+    (type) => (type.subtype === "pgp-signature" ? "PGP Signature" : undefined),
+    (type) => (type.subtype === "rtf" ? "Rich Text (RTF)" : undefined),
+    (type) => (type.subtype === "ogg" ? "Media (OGG)" : undefined),
+    (type) => (type.subtype === "octet-stream" ? "Binary Data" : undefined),
+    () => "Application/Binary Data",
+  ],
+  [Type.AUDIO]: [
+    ({ subtype }) => {
+      switch (subtype) {
+        case "mpeg":
+          return "Audio (MP3)";
+        case "mp4":
+          return "Audio (MP4)";
+        case "aac":
+          return "Audio (AAC)";
+        case "ogg":
+          return "Audio (OGG)";
+      }
+    },
+    () => "Audio",
+  ],
+  [Type.EXAMPLE]: [() => UNRECOGNIZED],
+  [Type.FONT]: [() => "Font"],
+  [Type.IMAGE]: [
+    (type) => {
+      switch (type.subtype) {
+        case "gif":
+          return "Image (GIF)";
+        case "heic":
+          return "Image (HEIC)";
+        case "jpeg":
+          return "Image (JPEG)";
+        case "png":
+          return "Image (PNG)";
+        case "tiff":
+          return "Image (TIFF)";
+        case "svg+xml":
+          return "Image (SVG)";
+        case "bmp":
+          return "Image (BMP)";
+        case "webp":
+          return "Image (WebP)";
+        case "x-icon":
+          return "Image (Icon)";
+        case "vnd.microsoft.icon":
+          return "Image (Icon)";
+        case "vnd.adobe.photoshop":
+          return "Image (Photoshop)";
```

Comment:
I have a refactoring question/suggestion related to capturing the type and subtype, which I like as `type (subtype)`. I'm noticing you're writing the type a lot without much change for example here. Would it be a lot of extra work to capture the type first (could be captured a variable or even a separate helper function with a switch statement) and then provide the subtype? I understand there are some cases where you want to only provide the subtype or suffix to be returned so this might add some complication.

Reply:
I don't understand what you're asking for... Unfortunately, we can't just assume that we can programatically pull a useful name consistently. Consider SVG or Photoshop above. This code might be repetitive but it is explicit and it is correct. There won't be subtle bugs where _some_ media types display incorrectly.

Reply:
What I'm asking is instead of writing the type ("Image"/"Audio"/"Video"/etc.) before each subtype, capturing this elsewhere and concatenating this with the subtype.

> Unfortunately, we can't just assume that we can programatically pull a useful name consistently. 

Although, I think this answers my question.

Reply:
Even if we do something like the following, it will be wrong more often that it's worth:

function toTitleCase(s: string): string {
  return s.charAt(0).toUpperCase() + s.slice(1);
}
function normalizeSubtype(m: MediaType): string {
  // Guess how the fuck the media type might look and hope to god we guess
  // correctly when it comes to brand names and that they don't capitalize
  // the whole thing or a word part-way through.
  // We could maintain a list of exceptions but is that really any better
  // than where we are now?
  if (m.subtype.startsWith("vnd.")) {
    const lastBit = s.split(".").pop();
    return toTitleCase(lastBit);
  }
  // Hopefully this file type is all uppercase! We're not maintaining any
  // overrides. So we'll definitely get MP3 wrong.
  return m.subtype.split(".").pop().toUpperCase();
}
function display(m: MediaType): string {
  return toTitleCase(m.type) + " " + "(" + normalizeSubtype(m) + ")";
}
```

Reply:
Even if we do something like the following, it will be wrong more often that it's worth:

```
function toTitleCase(s: string): string {
  return s.charAt(0).toUpperCase() + s.slice(1);
}
function normalizeSubtype(m: MediaType): string {
  // Guess how the media type might look and hope we guess
  // correctly when it comes to brand names and that they don't capitalize
  // the whole thing or a word part-way through.
  // We could maintain a list of exceptions but is that really any better
  // than where we are now?
  if (m.subtype.startsWith("vnd.")) {
    const lastBit = s.split(".").pop();
    return toTitleCase(lastBit);
  }
  // Hopefully this file type is all uppercase! We're not maintaining any
  // overrides. So we'll definitely get MP3 wrong.
  return m.subtype.split(".").pop().toUpperCase();
}
function display(m: MediaType): string {
  return toTitleCase(m.type) + " " + "(" + normalizeSubtype(m) + ")";
}
```